### PR TITLE
Update MemberList to reflect changes for invite permission change

### PR DESCRIPTION
**Pull request by jaiwanth-v**
The invite button doesn't reflect any changes when the invite permission is changed for the user viewing the member list. This PR fixes that.

Signed-off-by: Jaiwanth <jaiwanth2011@gmail.com>

### CODE REVIEW

File: src/components/views/rooms/MemberList.js [hunk 1]
```diff
@@ -196,6 +196,9 @@ export default class MemberList extends React.Component {
             event.getType() === "m.room.third_party_invite") {
             this._updateList();
         }
+        if (event.getContent().invite !== event.getPrevContent().invite) {
+            this.forceUpdate();
```
Comment (t3chguy) [inline src/components/views/rooms/MemberList.js:200]: `forceUpdate` is a smell, `canJoin` should be moved into state.
Additionally keep in mind this isn't the only thing that can change whether or not you can invite someone, if your power level changes that will affect it too

Comment (jaiwanth-v) [reply src/components/views/rooms/MemberList.js:200]: > `canJoin` should be moved into state.

Oh yes, that's a better idea. 
> if your power level changes that will affect it too

Forgot about that. Will modify.

**Comment by t3chguy** [comment]
Test failure looks to be a missing mock; you'll need to add `getMember: jest.fn(),` to test-utils mkStubRoom > currentState

**Comment by jaiwanth-v** [comment]
Still failing with the same error

**Comment by t3chguy** [comment]
Argh, my bad. MemberList-test doesn't use test-utils for that, feel free to leave that mock in but you'll need to add similar here https://github.com/matrix-org/matrix-react-sdk/blob/46bfbbadf9e448b795610c5bbe28da4d5bc21e83/test/components/views/rooms/MemberList-test.js#L90

File: src/components/views/rooms/MemberList.js [hunk 2]
```diff
@@ -136,11 +136,13 @@ export default class MemberList extends React.Component {
     _getMembersState(members) {
         // set the state after determining _showPresence to make sure it's
         // taken into account while rerendering
+        const cli = MatrixClientPeg.get();
         return {
             loading: false,
             members: members,
             filteredJoinedMembers: this._filterMembers(members, 'join'),
             filteredInvitedMembers: this._filterMembers(members, 'invite'),
+            canInvite: cli.getRoom(this.props.roomId).canInvite(cli.getUserId()),
```
Comment (t3chguy) [inline src/components/views/rooms/MemberList.js:145]: would be good to null-guard the room, other places in the code seem to as well

File: src/components/views/rooms/MemberList.js [hunk 3]
```diff
@@ -196,6 +198,10 @@ export default class MemberList extends React.Component {
             event.getType() === "m.room.third_party_invite") {
             this._updateList();
         }
+
+        const cli = MatrixClientPeg.get();
+        const canInvite = cli.getRoom(this.props.roomId).canInvite(cli.getUserId());
```
Comment (t3chguy) [inline src/components/views/rooms/MemberList.js:203]: would be good to null-guard the room, other places in the code seem to as well

File: src/components/views/rooms/MemberList.js [hunk 4]
```diff
@@ -196,6 +198,10 @@ export default class MemberList extends React.Component {
...
+        if (canInvite !== this.state.canInvite) this.setState({canInvite});
```
Comment (t3chguy) [inline src/components/views/rooms/MemberList.js:204]:
```suggestion
        if (canInvite !== this.state.canInvite) this.setState({ canInvite });
```

File: src/components/views/rooms/MemberList.js (hunk 2 above)
Comment (jaiwanth-v) [reply src/components/views/rooms/MemberList.js:145]: Okay I'll modify

**Review by t3chguy — APPROVED** [review]
LGTM - thanks

**Pull request merged**